miniupnpd/miniupnpd.c

```diff
@@ -1267,6 +1267,17 @@ init(int argc, char * * argv, struct runtime_vars * v)
 				break;
 #endif	/* ENABLE_MANUFACTURER_INFO_CONFIGURATION */
 #ifdef USE_NETFILTER
+#ifdef USE_NFTABLES
+			case UPNPFORWARDCHAIN:
+				set_rdr_name(FORWARD_CHAIN_NAME, ary_options[i].value);
+				break;
+			case UPNPNATCHAIN:
+				set_rdr_name(NAT_CHAIN_NAME, ary_options[i].value);
+				break;
+			case UPNPNATPOSTCHAIN:
+				set_rdr_name(NAT_POSTROUTING_CHAIN_NAME, ary_options[i].value);
+				break;
+#else
```

Comment:
the same modification can apply to standard iptables, right ?
I think miniupnpd_forward_chain/miniupnpd_nat_chain/miniupnpd_nat_postrouting_chain are only used in iptcrdr.c

Reply:
Yes, that's the intent. I just haven't been able to test the IPtables version, so I limited what I changed to what I could test easily. More a matter of time than anything else.

The way nat works on nftables, you must have both the prerouting and postrouting chains defined (to establish both inbound and outbound paths) irrespective of whether both chains have rules. So even though no rules are added to 'postrouting', it must be present or the rules in 'prerouting' won't work properly.
 
(part of my nefarious plan to implement privilege separation ;)